java/core/src/main/java/com/microsoft/bond/NullableBondType.java

```diff
@@ -120,6 +120,20 @@ protected final TValue deserializeValue(TaggedDeserializationContext context) th
     }
 
     @Override
+    protected final TValue deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        if (count == 0) {
+            return null;
+        } else if (count == 1) {
+            return this.valueType.deserializeValue(context);
+        } else {
+            // throws
+            Throw.raiseNullableListValueHasMultipleElementsDeserializationError(this.getFullName());
+            return null;
+        }
```

Comment:
Please follow the pattern used in the tagged version (specifically a single return statement). Please note that you are not calling readContainerEnd here, which is a bug.